### DISCUSSION TITLE
Errors - Try to access to my API

### DISCUSSION THREAD

**Discussion by SaninfomaxUN** [vercel/community#2601] [Help] [answered]
### Summary

Hi. I am trying to deploy my project (React - Node js - Express) where I have two servers (Front & Back), so I deployed them separately. The front and back unfolded well. However, I can't find any way that the front can call the api.
I tried calling directly by the address generated by vercel. However, it causes many problems, such as the following:

`Access to fetch at 'https://ecommerce-un-server.vercel.app/api/loginShopper' from origin 'https://ecommerce-un.vercel.app' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: It does not have HTTP ok status.`

And sometimes:
`message: 'Request failed with status code 504', name: 'AxiosError', code: 'ERR_BAD_RESPONSE', config: {…}, request: XMLHttpRequest,`

The API makes queries to a database stored in another separate hosting.
What can I do to change and/or fix these errors?
I appreciate your help. It is my first deploy.

### Steps to Reproduce

My App.js

`const routes = require("./Routes/Router.js")`
`const ConnectionDB = require("./Database/ConnectionDB.js")`

`const express = require("express");`
`const bodyParser = require('body-parser');`
`const cors = require('cors');`

`const app = express();`
`const port = 5000;`

`app.use(bodyParser.urlencoded({ extended: false }));`
`app.use(bodyParser.json());`
`app.use(cors({ origin: true }));`
`app.use(express.json());`

`app.use('/api',  routes)`

`app.get('/', (req, res) => {`
 ` res.send('Hey this is my API running 🥳')`
`})`

`app.listen(port, () => {`
 ` console.log("Server listening at http://localhost:${port}"); `
  `ConnectionDB.testConnection()`
`});`

Example of a call to API (from the Front):
`axios.post('https://ecommerce-un-server.vercel.app/api/check2FA', codeDataToSend)`
      `      .then(`
             `   res => {`
         `           setIsCode2FACorrect(!res.data)`
          `          checkVerification2FA(res.data)`

                `}`
            ` )`
           ` .catch(err => console.log(err));`

**Comment by amyegan** [comment, marked as answer]
It sounds like the frontend is blocked from the separate API backend. You should read about [Cross-Origin Resource Sharing (CORS)](https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS). 

If the backend is also hosted on Vercel, we have a guide to help [enable CORS on a Vercel project](https://vercel.com/guides/how-to-enable-cors). If the backend is hosted elsewhere, try enabling CORS in your code or look into any settings on your API host that would allow calls to be made from your frontend domain.

**Reply by SaninfomaxUN** [reply]
Thank you very much for your answer. The error of `has been blocked by CORS policy` has disappeared. :)
